### PR TITLE
Improvements for NRF52

### PR DESCRIPTION
**Pull request by raiden00pl**

### CODE REVIEW

File: arch/arm/src/nrf52/nrf52_lowputc.c [hunk 1]
```diff
@@ -216,7 +216,7 @@ void nrf52_usart_configure(uintptr_t base, const struct uart_config_s *config)
  ****************************************************************************/
 
 #ifdef HAVE_UART_DEVICE
-void nrf52_usart_disable(uintptr_t base)
+void nrf52_usart_disable(uintptr_t base, const struct uart_config_s *config)
```
Comment (acassis) [inline arch/arm/src/nrf52/nrf52_lowputc.c]: Mateuz, I think it is a good idea to add FAR to all these "const struct uart_config_s *", what do you think?

Comment (raiden00pl) [reply arch/arm/src/nrf52/nrf52_lowputc.c]: Yes, you are right. Fixed

**Comment by acassis** [comment]
Thank you Mateusz, please verify if everything is fine. Every nice improvements! I think now we can run NuttX on PineTime Watch and on other Watches/Bands powered by nRF52

**Comment by raiden00pl** [comment]
I think we still need support for SPI and GPIO interrupts to do something useful. I'll add them next week.

**Comment by patacongo** [comment]
@acassis FAR is not normally used in ARM code.  It is only required in the common code where it needs to have a value for some architectures.  It is fine in ARM code to assume that FAR is defined to be nothing.  Everyone assumes that.

**Comment by acassis** [comment]
@patacongo You are right! I saw a FAR in a function that he submitted and assumed it should be presented for all functions. My fault!

**Comment by patacongo** [comment]
@acassis Well, it should be used consistently in ARM code.  Either always user FAR or never use it in an ARM file.

**Comment by raiden00pl** [comment]
At the moment in arch/nrf52 in some places there is FAR in some not. So for now it is far from consistent. But I think it is also true for other ARM chips. 
If there is no strict rule (always use FAR vs never use FAR), then there will be not consistency.

**Comment by acassis** [comment]
Well, it appears at least 15182 inside arch/arm:
alan@dev:~/incubator-nuttx/arch/arm$ grep -r "FAR " | wc -l
15182

**Comment by patacongo** [comment]
Don't be tempted to change that with a simple substitution.  That would mess up the alignment.  It would require lots of manual effort or a very intelligent script.  It looks like Xiao Xiang must have had such an intelligent script when he removed all of the (void).  He also corrected all of the spacing that the that changed because of that.

**Comment by xiaoxiang781216** [comment]
I don't have the magic script either:(: only single line functions are replaced by the command automatically, other functions spread the multiple lines are done manually.

**Pull request merged**